models/article.model.js

```diff
@@ -1,18 +1,39 @@
 const db = require('../db/connection');
 
 
+
 //ARTICLES ENDPOINT
-exports.selectArticles = ()=>{
-   const queryString = 
+exports.selectArticles = (topicQuery,sort_by ='created_at',order = 'desc')=>{
+    validSortByQueries = ['title','topic','author','created_at','votes','comment_count'];
+    validOrderByQueries = ['asc','desc']
+    
+    if(!validSortByQueries.includes(sort_by) || !validOrderByQueries.includes(order)){
+        return Promise.reject({ status:400, msg: 'Bad Request'})
+    }
+    const queryString = 
      `SELECT articles.article_id,articles.title,articles.topic,articles.author,articles.created_at,articles.votes,count(comments.article_id=articles.article_id) AS comment_count
      FROM articles
      LEFT JOIN comments ON articles.article_id=comments.article_id
-     GROUP BY articles.article_id
-     ORDER BY articles.created_at desc;`
-   return db
+     GROUP BY articles.article_id 
+     ORDER BY ${sort_by} ${order};`;
+    
+    return db
     .query(queryString).then((result)=>{
-        return result.rows;
-    });
+        const allArticles = result.rows
+
+        if(topicQuery == undefined){
+        return allArticles;
+    }else{
+        const relevantArticles =  allArticles.filter(function(eachArticle){
```

Comment:
While filter works, it causes a more strain on the server.

Please could you have a look to see how you could incorporate this into your DB query.